script.js

```diff
@@ -1,4 +1,194 @@
+var generateOutcome = function () {
+  var randomInt = Math.floor(Math.random() * 3);
+  var map = { 0: "scissors", 1: "paper", 2: "stone" };
+  var outcome = map[randomInt];
+  return outcome;
+};
+
+// Create global variables
+var gameState = "NameInput";
+var playerName = "Player";
+var wins = 0;
+var loss = 0;
+var games = 0;
+var previousWinner = "no one";
+
+// create outcome variables
+var win = "Congratulations, you win! Want a rematch?";
+var lose = "Bummer, you lost. Try again next time!";
+var tie = "Its a draw. We'll have to go again";
+var invalid = `This is a game of scissors paper stone. Accepted input values are "scissors", "paper", "stone" or "random".`;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (gameState == "NameInput") {
+    // Start getting user input
+    playerName = input;
+    var myOutputValue = `Welcome to play the game ${playerName}. <br> Please select the following game modes: "Normal", "Reversed" or "Korean".`;
+    gameState = "GameSelect";
+    return myOutputValue;
+  } else if (gameState == "GameSelect") {
+    // Reset counters for existing game mode.
+    wins = 0;
+    loss = 0;
+    games = 0;
+    var gameSelect = input;
+    if (
+      gameSelect == "Normal" ||
+      gameSelect == "Reversed" ||
+      gameSelect == "Korean"
+    ) {
+      gameState = gameSelect;
+      myOutputValue = `Good choice ${playerName}! We will now proceed to play ${gameSelect} version of scissors, paper, stone! Best of Luck!`;
+    } else {
+      myOutputValue = `Sorry ${playerName}, this is an invalid entry. <br> Please select the following game modes: "Normal", "Reversed" or "Korean".`;
+    }
+    return myOutputValue;
+  } else if (gameState == "Normal") {
+    // check user and computer inputs
+    if (input == "random") {
+      var userChoice = generateOutcome();
+    } else {
+      var userChoice = input;
+    }
+
+    var computerOutcome = generateOutcome();
+    console.log("Input: " + userChoice);
+    console.log("Generated: " + computerOutcome);
+
+    // the default outcome of all games shall be invalid
+    var outcome = invalid;
+
+    // If else logic for normal mode
+    if (
+      (userChoice == "scissors" && computerOutcome == "stone") ||
+      (userChoice == "paper" && computerOutcome == "scissors") ||
+      (userChoice == "stone" && computerOutcome == "paper")
+    ) {
+      outcome = lose;
+      loss = loss + 1;
```

Comment:
a convenient syntax for this would be loss +=1 which would be the same as loss = loss +1